src/network/networkd-routing-policy-rule.c

```diff
@@ -0,0 +1,904 @@
+                }
+
+                if (rule->network->manager) {
+                        set_remove(rule->network->manager->rules, rule);
+                        set_remove(rule->network->manager->rules_foreign, rule);
+                }
+        }
+
+        free(rule);
+}
+
+static void routing_policy_rule_hash_func(const void *b, struct siphash *state) {
+        const RoutingPolicyRule *rule = b;
+
+        assert(rule);
+
+        siphash24_compress(&rule->family, sizeof(rule->family), state);
+
+        switch (rule->family) {
+        case AF_INET:
+        case AF_INET6:
+
+                if (!in_addr_is_null(rule->family, &rule->from)) {
+                        siphash24_compress(&rule->from, FAMILY_ADDRESS_SIZE(rule->family), state);
+                        siphash24_compress(&rule->from_prefixlen, sizeof(rule->from_prefixlen), state);
+                }
+
+                if (!in_addr_is_null(rule->family, &rule->to)) {
+                        siphash24_compress(&rule->to, FAMILY_ADDRESS_SIZE(rule->family), state);
+                        siphash24_compress(&rule->to_prefixlen, sizeof(rule->to_prefixlen), state);
+                }
+
+                siphash24_compress(&rule->tos, sizeof(rule->tos), state);
+                siphash24_compress(&rule->fwmark, sizeof(rule->fwmark), state);
+                siphash24_compress(&rule->table, sizeof(rule->table), state);
+
+                break;
+        default:
+                /* treat any other address family as AF_UNSPEC */
+                break;
+        }
+}
+
+static int routing_policy_rule_compare_func(const void *_a, const void *_b) {
+        const RoutingPolicyRule *a = _a, *b = _b;
+
+        if (a->family < b->family)
+                return -1;
+        if (a->family > b->family)
+                return 1;
+
+        switch (a->family) {
+        case AF_INET:
+        case AF_INET6:
+                if (a->from_prefixlen < b->from_prefixlen)
+                        return -1;
+                if (a->from_prefixlen > b->from_prefixlen)
+                        return 1;
+
+                if (a->to_prefixlen < b->to_prefixlen)
+                        return -1;
+                if (a->to_prefixlen > b->to_prefixlen)
+                        return 1;
+
+                if (a->tos < b->tos)
+                        return -1;
+                if (a->tos > b->tos)
+                        return 1;
+
+                if (a->fwmask < b->fwmark)
+                        return -1;
+                if (a->fwmask > b->fwmark)
+                        return 1;
+
+                if (a->table < b->table)
+                        return -1;
+                if (a->table > b->table)
+                        return 1;
+
+                if (!in_addr_is_null(a->family, &a->from))
+                        return memcmp(&a->from, &b->from, FAMILY_ADDRESS_SIZE(a->family));
+
+                if (!in_addr_is_null(a->family, &a->to))
```

Comment:
The code sample you pasted is garbled. Can you paste how you think it should look?